specs/waku/v2/waku-store.md

```diff
@@ -2,36 +2,88 @@
 title: Waku
 version: 2.0.0-beta1
 status: Draft
-authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>
+authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>, Sanaz Taheri <sanaz@status.im>
 ---
 
 # Table of Contents
 
 - [Abstract](#abstract)
+  * [Waku Store Protocol](#Waku-Store-Protocol)
+  * [Pagination](#Pagination)
 - [Wire Specification](#wire-specification)
   * [Protobuf](#protobuf)
 - [Changelog](#changelog)
 - [Copyright](#copyright)
 
 # Abstract
 
+## Waku Store Protocol 
 `WakuStore` is a protocol to enable querying of messages received through relay protocol and stored by other nodes.
 
+## Pagination
```

Comment:
I'd probably pull this down to under wire specification or as part of it. So it'd be something like:

Abstract
Motivation (currently missing and probably out of scope for PR)
Wire spec
 - details about fields
   - possibly on two levels if needed

E.g. something like this https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-relay.md#rpc

Reply:
Right now it seems like it is a lot of duplication, I might be wrong and we need a separate section (e.g. before or after wire spec), but I think it can be done as part of it.

Reply:
If we want to add motivation, it'd be something about nodes being offline a lot and wanting to get historical messages, etc

Reply:
Sure, I'll include the motivation, perhaps it should cover motivation for the store protocol as well as the pagination, right?

Reply:
In the abstract section, my aim was to provide an overview of the pagination process and to give an insight into 
why the protobuf messages that are provided in the wire spec are even needed. I thought it would be easier to understand the whole spec this way. However, I agree that the current writeup has duplications, and the wire specs or the abstract must be shorten.

Reply:
I see. To me, the abstract should be quite concise and max 200 words (probably less in this case because it isn't a very complex protocol) without too much technical detail. If that can be achieved without detailing implementation details it seems fine to have a sentence or three re pagination in there.

Reply:
Kind of like an executive summary to get a high level overview of what the store protocol is and why one should care

Reply:
> I'd probably pull this down to under wire specification or as part of it. So it'd be something like:
> 
> Abstract
> Motivation (currently missing and probably out of scope for PR)
I will open a PR for the motivation